src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,297 @@
+        // In practice, there are many ways one can use to select
+        // the i and j. In this simplified code, we select them randomly.
+        size_t j = rand() % data.n_cols;
+        while (j == i)
+        {
+          j = rand() % data.n_cols;
+        }
+
+        // Calculate Ej = f(x(j)) - y(j) using (2).
+        E(j) = intercept + arma::sum(alpha.t() % trainLabels % K.col(j).t())
+               - trainLabels(j);
+
+        // Saving old alpha values.
+        double alpha_i_old = alpha(i);
+        double alpha_j_old = alpha(j);
+
+        // Compute L and H to find max and min values of alphas.
+        if (trainLabels(i) == trainLabels(j))
+        {
+          L = std::max(0.0, alpha(j) + alpha(i) - regularization);
+          H = std::min(regularization, alpha(j) + alpha(i));
+        }
+        else
+        {
+          L = std::max(0.0, alpha(j) - alpha(i));
+          H = std::min(regularization, regularization +
+                       alpha(j) - alpha(i));
+        }
+
+        if (L == H)
+          continue;
+
+        // Compute eta by (14).
+        eta = 2 * K(i, j) - K(i, i) - K(j, j);
+        if (eta >= 0)
+          continue;
+        // Compute and clip new value for alpha j using (12) and (15).
+        alpha(j) = alpha(j) - (trainLabels(j) * (E(i) - E(j))) / eta;
+
+        // Clip.
+        alpha(j) = std::min(H, alpha(j));
+        alpha(j) = std::max(L, alpha(j));
+
+        // Check if change in alpha is noticeable or not.
+        if (std::abs(alpha(j) - alpha_j_old) < tol)
+        {
+          alpha(j) = alpha_j_old;
+          continue;
+        }
+
+        // Determine value for alpha i using (16).
+        alpha(i) = alpha(i) + trainLabels(i) * trainLabels(j)
+                   * (alpha_j_old - alpha(j));
+
+        // Compute b1 and b2 using (17) and (18) respectively.
+        double b1 = intercept - E(i)
+                    - trainLabels(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - trainLabels(j) * (alpha(j) - alpha_j_old) *  K(i, j);
+        double b2 = intercept - E(j)
+                    - trainLabels(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - trainLabels(j) * (alpha(j) - alpha_j_old) *  K(j, j);
+
+        // Compute b by (19).
+        if (0 < alpha(i) && alpha(i) < regularization)
+          intercept = b1;
+        else if (0 < alpha(j) && alpha(j) < regularization)
+          intercept = b2;
+        else
+          intercept = (b1 + b2) / 2;
+
+        num_changed_alphas = num_changed_alphas + 1;
+      }
+    }
+
+    if (num_changed_alphas == 0)
+      count = count + 1;
+    else
+      count = 0;
+  }
+  // Calculating paramter values for linear kernel.
+  parameters = (data * (alpha.t() % trainLabels).t()).t();
+  double threshold = arma::as_scalar(arma::mean(alpha));
+
+  // Saving values of labels and sample data to be used
+  // with kernel function.
+  savedLabels = arma::zeros(1, data.n_cols);
+  trainingData = arma::mat(data);
```

Comment:
can you remove `arma::mat` from here?

Reply:
Also can you add the change in a separate commit (rather than force pushing) so, that I can see exactly what causes the build failure?

Reply:
> can you remove `arma::mat` from here?

I have changed only this.


> Also can you add the change in a separate commit (rather than force pushing) so, that I can see exactly what causes the build failure?

Sure I just don't want extra commits I will take care of it next time :)

Reply:
> Sure I just don't want extra commits I will take care of it next time :)

You could try to squash the previous commits using interactive rebase and clear the commit history. 
Force pushing makes it hard to see the changes.